### DISCUSSION TITLE
Rendering two charts/tables side by side

### DISCUSSION THREAD

**Discussion by lorefnon** [sqlpage/SQLPage#168] [Ideas]
I think it would be useful to have some generic way to control the layout of chart/table panels. I was not able to figure out from docs and a cursory examination of the templates but maybe there is some trick already?

I was initially looking to embed charts within cards which do seem to have multi-column support but it looks like components can't be composed (yet?).

**Comment by lovasoa** [comment]
Hello ! 

  Indeed, there is no way to compose components or to create complex layouts at the moment, and it is not on the roadmap. You can always create your [own custom components](https://sql.ophir.dev/custom_components.sql) if you want, of course.

  If you do have an idea for how sqlpage could add multi-column layouts without becoming more complicated to use, I'm open to discussion :)

**Comment by lorefnon** [comment]
One solution I can think of is to have a row level parameter in card like
embed=somefile.sql that embeds the rendered output of another file (without
the shell of course) within the card.

I am not sure if it would fall within your acceptable complexity criteria
but I think some flexibility on the layouts will benefit many users.

It's been a while I have done anything complex with handlebars but I can
take a stab at this if you find the feature acceptable.

The issue with creating a custom component is that I don't seem to have a
good way to create a container and reuse your built in components within
them.

**Reply by lovasoa** [reply]
Yes, I think that would be acceptable !

```sql
select 'card' as component;
select 'other.sql?id=' || id as embed from my_table;
```

But this isn't trivial to implement, and you'll need more than just handlebars.

**Reply by lovasoa** [reply]
If we implement that with `<iframes>`, then it's way easier, but it may be less appealing for the user...
